nexus/reconfigurator/planning/src/blueprint_builder/builder.rs

```diff
@@ -122,8 +131,19 @@ pub enum Ensure {
 /// actions taken or no action was necessary
 #[derive(Debug, Clone, Copy, Eq, PartialEq)]
 pub enum EnsureMultiple {
-    /// action was taken, and multiple items were added
-    Changed { added: usize, removed: usize },
+    /// action was taken within the operation
+    Changed {
+        /// An item was added to the blueprint
+        added: usize,
+        /// An item was updated within the blueprint
+        updated: usize,
+        /// An item was expunged in the blueprint
+        expunged: usize,
+        /// An item was removed from the blueprint.
+        ///
+        /// This usually happens after the work of expungment has completed.
+        removed: usize,
```

Comment:
Sooooo this is kinda important for tracking dataset state with a blueprint:

- Datasets can be added (see: `added`)
- Datasets can be modified -- maybe we want to change their quota? (see: `updated`)
- Datasets can also be **expunged**, which means we want to keep a record of their expungement. This is distinct from actually **removing** the dataset from the blueprint, which can only happen once expungement completes later.

Hence, I split this into a few additional states.